app/helpers/holdings_helper.rb

```diff
@@ -25,11 +25,35 @@ def fetch_sfx_holdings(document)
     end
   end
   
+  # TODO: If the way that libraries are mapped changes (replacing config/location.yml) this should follow that scheme
+  UAL_SHIELD_LIBRARIES = [
+    SYMPHONY_LIBRARY_LOCATIONS[:uainternet],
+    SYMPHONY_LIBRARY_LOCATIONS[:neosfree]
+  ].freeze
+  
+  # determines whether or not ual shield should be displayed for the item
   def display_ual_shield(document)
     return false unless document["location_tesim"]
-    return true if document["location_tesim"].include?(I18n.t('ual_shield_UAL'))
-    return true if document["location_tesim"].include?(I18n.t('ual_shield_NEOS'))
+    UAL_SHIELD_LIBRARIES.each { |library| return true if document["location_tesim"].include? library }
     false
   end
 
+  # TODO: If the way that libraries are identified changes (replacing config/location.yml) this should follow that scheme
+  READ_ON_SITE_LOCATION_RCRF = 'UARCRF'.freeze # Research and Collections Resource Facility
+  READ_ON_SITE_LOCATION_BPSC = 'UASPCOLL'.freeze # Bruce Peel Special Collections
+  
+  # returns path to read on site form
+  # TODO: replace placeholder with appropriate path_for
+  def read_on_site_path(item)
+    return '/#TODO_RCRF' if item[:location] == READ_ON_SITE_LOCATION_RCRF
+    return '/#TODO_BPSC' if item[:location] == READ_ON_SITE_LOCATION_BPSC
+    errors_unprocessable_path
```

Comment:
It would be nice if you could never reach the `errors_unprocessable_path`? If things go bad, why give the user a link to a 422 page?

You could move logic into the view maybe and avoid this entirely?

```
<% if item[:status] == 'READONSITE' %>
  <% if item[:location] == READ_ON_SITE_LOCATION_RCRF %>
      <%= link_to "Request Me",'/#TODO_RCRF', {:id => 'requestLink', :class => 'lightboxLink'} %>
  <% elsif item[:location] == READ_ON_SITE_LOCATION_BPSC %>
     <%= link_to "Request Me", '/#TODO_BPSC', {:id => 'requestLink', :class => 'lightboxLink'} %>
  <% end%>
<%- end -%>
```

Not pretty but gets the job done without any weird side effects like linking to a 422 page (it just doesn't show a link if its not one of the locations)

Reply:
I wasn't sure what a good default would be. Perhaps it should be
```suggestion
    nil
```
Then the view can be
```
                    <%= link_to_unless "Request Me", read_on_site_path(item), {:id => 'requestLink', :class => 'lightboxLink'} %>
```